docs/topics/migrations.txt

```diff
@@ -467,6 +467,74 @@ You can pass a second callable to
 want executed when migrating backwards. If this callable is omitted, migrating
 backwards will raise an exception.
 
+.. _data_migrations_and_multiple_databases:
+
+Data migrations and multiple databases
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One of the challenges that can be encountered is figuring out whether or
+not to run a migration when using multiple databases.
+
+For example, you may want to **only** run a migration against a particular
+database.
+
+In order to do that you can filter inside of your data migration
+by looking at the ``schema_editor.connection.alias`` attribute.
+However it is better to have your ``dbrouter`` leverage the same
+``allow_migrate`` call by making the following method in your
+router::
+
+        class MyRouter(object):
+
+            def allow_migrate(self, db, model):
+                # your typical migration code here
+
+            def runpython_ok(self, apps, schema_editor, model):
+                db = schema_editor.connection.alias
+                if self.allow_migrate(db, model):
+                    return True
+                return False
+
+Then to leverage this in your migrations, do the following::
+
+        # -*- coding: utf-8 -*-
+        from __future__ import unicode_literals
+
+        from django.db import models, migrations
+        from yourappname.dbrouter import MyRouter
+        import sys
+
+        def forwards(apps, schema_editor):
+
+            # only run this for the correct databases
+            YourModel = apps.get_model("yourappname", "YourModel")
+            if not MyRouter().runpython_ok(apps, schema_editor, YourModel):
```

Comment:
You have two options here: either use `if self.allow_migrate(schema_editor.connection.alias, YourModel)` which effectively checks (https://github.com/django/django/blob/master/django/db/migrations/operations/base.py#L101-L111)
```python
return (
    not model._meta.proxy and
    not model._meta.swapped and
    model._meta.managed and
    router.allow_migrate(connection_alias, model)
)
```

Or access the router directly as you did. But this implies that the router is importable during the entire lifetime of the project. This in turn has to be documented.